### DISCUSSION TITLE
Failing to find font and soundfont files when hosting from a directory that isn't the root

### DISCUSSION THREAD

**Discussion by nimblespark** [CoderLine/alphaTab#1629] [Q&A] [answered]
I'm using vite for my application and I got this library working well in dev mode. However, when I deploy the vite build to my website, I'm getting a 404 on the font and soundfont files. The problem is that I'm hosting from a directory that isn't the root but the requests for the font files are being made to the root. 

For example, if the website I'm trying to host from is example.com/sandbox, it's trying to find example.com/font/Bravura.woff instead of example.com/_**sandbox**_/font/Bravura.woff.

I've tried changing the base path on the vite config file but that didn't work. I've also tried adding a base path tag on the index.html file but that didn't work either.
Does anyone have any ideas how I can configure it to look for these files in the correct spot?

**Comment by Danielku15** [comment, marked as answer]
You need to configure this setting according to your app:
https://www.alphatab.net/docs/reference/settings/core/fontdirectory

See also: 
https://github.com/CoderLine/alphaTabSamplesWeb/blob/main/src/vite-vue/src/App.vue#L12
https://github.com/CoderLine/alphaTabSamplesWeb/blob/main/src/vite-react/src/App.tsx#L13


The fontDirectory setting is used to generate a style tag with the web font definitions: https://github.com/CoderLine/alphaTab/blob/8d816be65516fef4cba5c187bde5e6e662628435/src/Environment.ts#L121

The soundFontUrl is used to load the file via Ajax: https://github.com/CoderLine/alphaTab/blob/8d816be65516fef4cba5c187bde5e6e662628435/src/platform/javascript/AlphaSynthWebWorkerApi.ts#L282

**Reply by nimblespark** [reply]
Thanks so much, that fixed it. Btw super cool library
